Source/Classes/PINProgressiveImage.m

```diff
@@ -116,9 +117,14 @@ - (CFTimeInterval)startTime
     return startTime;
 }
 
-- (void)updateProgressiveImageWithData:(NSData *)data expectedNumberOfBytes:(int64_t)expectedNumberOfBytes
+- (void)updateProgressiveImageWithData:(nonnull NSData *)data expectedNumberOfBytes:(int64_t)expectedNumberOfBytes isResume:(BOOL)isResume
 {
     [self.lock lock];
+        if (isResume) {
+            NSAssert(self.mutableData == nil, @"If we're resuming, data shouldn't be setup yet.");
+            self.startingBytes = data.length;
```

Comment:
Should we set the property with the lock held in this case? Does this class not follow like the naming convention yet: `_locked_...`?